pkg/controllers/user/networkpolicy/policy.go

```
@@ -0,0 +1,346 @@
+
+	err := programNetworkPolicy(pnplc.mgr, projectIDNoColon)
+	return pnp, err
+}
+
+func (pnplc *projectNetworkPolicyLifecycle) Remove(pnp *v3.ProjectNetworkPolicy) (*v3.ProjectNetworkPolicy, error) {
+	logrus.Infof("pnplc Remove pnp=%+v", pnp)
+
+	projectIDNoColon, ok := pnp.Labels[projectIDNoColonLabel]
+	if !ok {
+		return pnp, fmt.Errorf("couldn't find projectIDNoLabel for pnp=%+v", *pnp)
+	}
+	logrus.Debugf("pnplc Updated projectIDNoColon: %v", projectIDNoColon)
+
+	set := labels.Set(map[string]string{projectIDNoColonLabel: projectIDNoColon})
+
+	err := pnplc.mgr.cluster.K8sClient.NetworkingV1().NetworkPolicies("").DeleteCollection(nil, v1.ListOptions{LabelSelector: set.String()})
+	if err != nil {
+		logrus.Errorf("error deleting network policies for projectIDNoColon=%v", projectIDNoColon)
+		return pnp, err
+	}
+	return pnp, nil
+}
+
+//func (plc *projectLifecycle) Create(p *v3.Project) (*v3.Project, error) {
+//	logrus.Infof("plc Create: %+v", *p)
+//	plc.createDefaultNetworkPolicy(p)
+//	return p, nil
+//}
+//
+//func (plc *projectLifecycle) Updated(p *v3.Project) (*v3.Project, error) {
+//	logrus.Infof("plc Updated: %+v", *p)
+//	return p, nil
+//}
+//
+//func (plc *projectLifecycle) Remove(p *v3.Project) (*v3.Project, error) {
+//	logrus.Infof("plc Remove: %+v", *p)
+//	// Need to delete the network  policy?
+//	return p, nil
+//}
+//
+//func (plc *projectLifecycle) createDefaultNetworkPolicy(p *v3.Project) error {
+//	return plc.mgr.createDefaultNetworkPolicy(p)
+//}
+
+func (ps *projectSyncer) Sync(key string, p *v3.Project) error {
+	if p == nil {
+		return nil
+	}
+	return ps.mgr.createDefaultNetworkPolicy(p)
+}
+
+func (mgr *mgr) createDefaultNetworkPolicy(p *v3.Project) error {
+	if p == nil {
+		return nil
+	}
+	existingPolicies, err := mgr.pnpLister.List(p.Name, labels.NewSelector())
+	if err != nil {
+		logrus.Errorf("error fetching existing project network policy: %v", err)
+		return nil
```

why do you mask the error?